### PR TITLE
added code for uploading codecov report

### PR DESCRIPTION
**Pull request by simsong**
Actually upload the coverage report to codecov.
FInd the coverage report here: https://app.codecov.io/github/dfxml-working-group/dfxml_python

### CODE REVIEW

**Comment by simsong** [comment]
pretty:
image

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -6,15 +6,18 @@
 name: CI (python)
 on: [push, pull_request]
 
+env:
+  COVERAGE_OS: ubuntu-latst
```
Comment (ajnelson-nist) [inline .github/workflows/continuous-integration.yml:10]: Typo - should this be `ubuntu-latest`?

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -50,3 +53,11 @@ jobs:
 
       - name: Test with pytest
         run: pytest --cov=dfxml --cov-report=xml .
```
Comment (ajnelson-nist) [inline .github/workflows/continuous-integration.yml:55]: Am I understanding this right: The coverage reported on codecov only pertains to what is reported by `pytest`, from this command?

**Review by ajnelson-nist — CHANGES_REQUESTED** [review]
I think I only only have one change request, a typo fix.  I'd appreciate an answer on the `pytest` line, but it's non-blocking.  I otherwise approve.

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (simsong) [reply .github/workflows/continuous-integration.yml:10]: Yes. It was a typo after I had it working!

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (simsong) [reply .github/workflows/continuous-integration.yml:55]: Correct. When pytest runs with the --cov= option, it creates a coverage report, which basically outputs to a file every line of every python file that is executed. So the coverage command literally shows you what was covered. You can also add things to the coverage configuration file to exclude things from coverage based on patterns and what not, to make you look better (or so you don't look bad for things that you can't possibly cover).

Comment (ajnelson-nist) [reply .github/workflows/continuous-integration.yml:55]: Understood, thanks!

**Review by ajnelson-nist — APPROVED** [review]

**Comment by simsong** [comment]
change made. I will accept this and then add the badge.

**Comment by ajnelson-nist** [comment]
> change made. I will accept this and then add the badge.

Whoops, sorry.  I didn't know you wanted to add a badge too.

**Comment by simsong** [comment]
You can’t add a badge until it’s committed to main. Codecov bug.

**Pull request merged**